### DISCUSSION TITLE
Numlock in Wayland

### DISCUSSION THREAD

**Discussion by Qpernicus** [qtile/qtile#3992] [Q&A]
I was wondering (and can't find it) what the preferred way is to activate numlock under Wayland. I now it under sway/Wayland, and used to use numlockx but after switching my numlock is not on by default/always. Can I 'rewrite' the sway command in any way? Input:keyboard numlock on, or something like that. 

I rather keep this setting in the Qtile config and not somewhere in my arch/systems settings so I have control which settings get used with which WM.

**Comment by Crystal4276** [comment]
The following works with X11, no idea with wayland.

You could prepare a script autostart.sh that include numlockx.
And then use this hook that trigger it at qtile start
```python
import os
import subprocess
@hook.subscribe.startup_once
def autostart():
    home = os.path.expanduser('~/.config/qtile/autostart.sh')
    subprocess.Popen([home])
```

**Reply by Qpernicus** [reply]
Yeah well, perhaps I didn't make it clear. That's exactly what I have/had with Qtile on Xorg. But numlockx doesn't seem to work under Wayland so I'm looking voor an alternative
